packages/node-resolve/src/resolveId.js

```diff
@@ -0,0 +1,130 @@
+import fs from 'fs';
+import path from 'path';
+import { promisify } from 'util';
+
+import resolve from 'resolve';
+
+import { getPackageName } from './util';
+
+const resolveImportPath = promisify(resolve);
+const readFile = promisify(fs.readFile);
+
+const pathNotFoundError = (subPath, pkgPath) =>
+  new Error(`Package subpath '${subPath}' is not defined by "exports" in ${pkgPath}`);
+
+function findExportKeyMatch(keys, subPath) {
+  return keys.find((key) => (key.endsWith('/') ? subPath.startsWith(key) : key === subPath));
+}
+
+function mapSubPath(pkgJsonPath, subPath, key, value) {
+  if (typeof value === 'string') {
+    if (value.endsWith('/')) {
```

Comment:
Note that subpath patterns (`endsWith('/*')` here) are now the preferred form over the trailing `/`. See https://nodejs.org/dist/latest-v15.x/docs/api/packages.html#packages_subpath_patterns.

The trailing `/` is actually being deprecated in https://github.com/nodejs/node/pull/35747.

No stress getting this into this PR right now, but more just letting you know as I find it much nicer to work with and feedback is welcome further.

Reply:
Added this